### PR TITLE
Pearl

### PR DESCRIPTION
**Pull request by rogeredthat**
1. Look for any mistakes in the refactoring of `public/../portals.js`
2. Significant changes in register-button.jade and register-button.js could potentially break the system by corrupting the data. Review carefully. Test if possible.

### CODE REVIEW

**Comment by aero31aero** [comment]
Tracking review status. Commits approved:

- [x] - 1
- [ ] - 2
- [ ] - 3
- [ ] - 4
- [x] - 5

File: public/static/scripts/portal.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 var globalEventID = Date.now();
 var currentEditable;
 var description;
+var portal;
 (function () {
 	portal = $("#_portal");
-
+	portal.simpleTextFields = ["name", "tagline", "category", "tickets", "type", "venue", "startTime", "endTime", "price", "prize", "route", "teamSize"];
```
Comment (aero31aero) [inline public/static/scripts/portal.js]: I believe this logic can be made better, with simply adding that type argument that we have in form generator be added as a data attribute to all `field-*` elements. Then just a simple JQuery magic and you get all the fields of certain types.

Obviously, that code should go in a generic form handler eventually.

File: public/static/scripts/portal.js [hunk 2]
```diff
@@ -64,54 +64,17 @@ function closeEditor() {
...
-			$('#field-prize').val("");
-		if (data.route)
-			$('#field-route').val(data.route);
-		else
-			$('#field-route').val("");
-		if (data.teamSize != null)
-			$('#field-teamSize').val(data.teamSize);
-		else
-			$('#field-teamSize').val("");
+		portal.simpleTextFields.forEach(function(field) {
```
Comment (aero31aero) [inline public/static/scripts/portal.js]: This is already a lot cleaner than before. A nice improvement. Should be applied to other forms as well.

File: routes/api/services/events.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ var schema = new Schema({
 	body: String,
 	type: String,
 	category: String,
+	tickets: String,
```
Comment (aero31aero) [inline routes/api/services/events.js]: Do we need to do a migration for this or do we wait for the concerned people to add this in production?

File: public/static/scripts/events.js [hunk 4]
```diff
@@ -9,9 +9,9 @@ $('document').ready(function () {
 				var elem = this;
 				$(elem).addClass('hidden');
 				var fields = [];
-				fields.push($(elem).find('span.tag').text().toLowerCase());
-				fields.push($(elem).find('h3.name').text().toLowerCase());
-				fields.push($(elem).find('span.tagline').text().toLowerCase());
+				fields.push($(elem).find('.tag').text().toLowerCase());
```
Comment (aero31aero) [inline public/static/scripts/events.js]: Correct me if I'm wrong here, but did we accidentally change the markup without checking and updating search? Good place to start with integration tests.

File: routes/api/services/events.js (hunk 3 above)
Comment (rogeredthat) [reply routes/api/services/events.js]: Migration wouldn't be necessary.

File: public/static/scripts/events.js (hunk 4 above)
Comment (rogeredthat) [reply public/static/scripts/events.js]: Exactly. That's what happened.

File: public/static/scripts/portal.js (hunk 1 above)
Comment (rogeredthat) [reply public/static/scripts/portal.js]: Yeah. But then I think the field suffix should be used carefully. I think the field-textarea which is the base element of the simpleMD editor would also be in there and create an extra entry. It needs more thought perhaps. Let's let this be for now.

**Pull request merged**